### PR TITLE
feat(zrepl): bumping the replica version

### PR DESCRIPTION
**Pull request by pawanpraka1**
As we have made rebuilding protocol changes, so bumping the replica verison here and also adding support for backward compatibility.

Signed-off-by: Pawan <pawan@mayadata.io>

### CODE REVIEW

File: src/data_conn.c [hunk 1]
```diff
@@ -955,8 +955,13 @@ uzfs_zvol_rebuild_dw_replica(void *arg)
 			}
 			continue;
 		}
-		ASSERT((hdr.opcode == ZVOL_OPCODE_READ) &&
-		    (hdr.flags & ZVOL_OP_FLAG_REBUILD));
+		if ((hdr.opcode != ZVOL_OPCODE_READ) ||
```
Comment (vishnuitta) [inline src/data_conn.c:958]: :+1:

File: src/data_conn.c [hunk 2]
```diff
@@ -1566,6 +1571,52 @@ uzfs_zvol_create_internal_snapshot(zvol_state_t *zv, zvol_state_t **snap_zv,
 	return (ret);
 }
 
+int
+hanlde_afs_transition(zvol_info_t *zinfo, zvol_rebuild_scanner_info_t *warg)
```
Comment (vishnuitta) [inline src/data_conn.c]: fn name spelling fix required

File: include/zrepl_prot.h [hunk 3]
```diff
@@ -39,7 +39,8 @@ extern "C" {
 
 #define	MIN_SUPPORTED_REPLICA_VERSION	3
 
-#define	REPLICA_VERSION	4
+#define	REPLICA_REBUILD_VERSION	5
```
Comment (vishnuitta) [inline include/zrepl_prot.h]: different name that helps to know the feature in rebuild..
for this case, like, MIN_RESIZE_REPLICA_VERSION

Comment (vishnuitta) [reply include/zrepl_prot.h]: might be MIN_RESIZE_REBUILD_VERSION

Comment (vishnuitta) [reply include/zrepl_prot.h]: or RESIZE_REBUILD_MIN_VERSION?

Comment (pawanpraka1) [reply include/zrepl_prot.h]: we have RESIZE op code and AFS_STARTED op code, so it should not say RESIZE specifically

Comment (pawanpraka1) [reply include/zrepl_prot.h]: on the other hand I think RESIZE_REBUILD_MIN_VERSION seems to best fit as AFS started is to fix a bug but RESIZE is a major feature. so going ahead with RESIZE_REBUILD_MIN_VERSION.

**Review by vishnuitta — APPROVED** [review]
changes are good

**Comment by vishnuitta** [comment]
merging this as this depends on https://github.com/openebs/cstor/pull/262

**Pull request merged**